src/components/FileList/FileList.js

```diff
@@ -20,17 +31,189 @@ const FileList = ({fileList, studyId}) => {
   const handlePageClick = (e, {activePage}) => {
     setPage(activePage);
   };
-  const sortedFileList = fileList
-    .filter(obj => fileLatestStatus(obj.node) === 'CHN')
-    .concat(fileList.filter(obj => fileLatestStatus(obj.node) !== 'CHN'));
-  const pageCount = Math.ceil(sortedFileList.length / perPage);
-  const pageItems = sortedFileList.slice(
+
+  const [sortMethod, setSortMethod] = useState('');
+  const [sortDirection, setSortDirection] = useState('ascending');
+  const [typeFilterStatus, setTypeFilterStatus] = useState('');
+  const [approvalFilterStatus, setApprovalFilterStatus] = useState('');
+  const [searchString, setSearchString] = useState('');
+  const [searchOption, setSearchOption] = useState('name');
+
+  const statusOptions = Object.keys(versionState).map(state => ({
+    key: state,
+    value: state,
+    text: versionState[state].title,
+    content: <Badge state={state} />,
+  }));
+  const typeOptions = Object.keys(fileTypeDetail).map(type => ({
+    key: type,
+    value: type,
+    text: fileTypeDetail[type].title,
+    content: (
+      <small>
+        <Icon name={`${fileTypeDetail[type].icon || 'question'}`} />
+        {' ' + fileTypeDetail[type].title}
+      </small>
+    ),
+  }));
+  const sortOptions = [
+    {
+      key: 'createDate',
+      value: 'createDate',
+      text: 'Create date',
+    },
+    {
+      key: 'modifyDate',
+      value: 'modifyDate',
+      text: 'Modified date',
+    },
+  ];
+  const searchOptions = [
+    {key: 'name', text: 'Title', value: 'name'},
+    {key: 'description', text: 'Description', value: 'description'},
+  ];
+
+  const sortedFileList = () => {
+    var sortedList = [];
+    sortedList = fileList;
+    if (
+      sortMethod === '' &&
+      typeFilterStatus === '' &&
+      approvalFilterStatus === ''
+    ) {
+      sortedList = fileList
+        .filter(obj => fileLatestStatus(obj.node) === 'CHN')
+        .concat(fileList.filter(obj => fileLatestStatus(obj.node) !== 'CHN'));
+    } else {
+      sortedList =
+        sortMethod === 'createDate'
+          ? createDateSort(fileList, sortDirection)
+          : modifiedDateSort(fileList, sortDirection);
+      if (typeFilterStatus !== '') {
+        sortedList = sortedList.filter(
+          obj => obj.node.fileType === typeFilterStatus,
+        );
+      }
+      if (approvalFilterStatus !== '') {
+        sortedList = sortedList.filter(
+          obj => fileLatestStatus(obj.node) === approvalFilterStatus,
+        );
+      }
+    }
+    if (searchString !== '') {
```

Comment:
I have a feeling there is a way to avoid checking the value of the filters. If you filter objects by an empty string, does it return all results, or none?